### PR TITLE
Adding StringLiteralParameter subclass

### PR DESCRIPTION
**Pull request by chris-j-tang**

### CODE REVIEW

File: src/Commands/Parameters/StringLiteralParameter.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * A string literal parameter.
+ */
+export class StringLiteralParameter extends Parameter {
+    /**
+     * The name of the string literal.
+     */
+    public name: string;
+
+    /**
+     * Initializes a new instance of the Parameter class.
```
Comment (JoshuaKGoldberg) [inline src/Commands/Parameters/StringLiteralParameter.ts]: `Parameter` typo

File: src/Commands/Parameters/StringLiteralParameter.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+     * 
+     * @param descriptor   A plain-text description of the parameter.
```
Comment (JoshuaKGoldberg) [inline src/Commands/Parameters/StringLiteralParameter.ts]: Check the docs - copypasta

File: src/Commands/Parameters/StringLiteralParameter.ts (hunk 1 above)
Comment (chris-j-tang) [inline src/Commands/Parameters/StringLiteralParameter.ts]: I still haven't found the parameter typo -- would you mind pointing me to which one is misspelled?

File: src/Commands/Parameters/StringLiteralParameter.ts (hunk 2 above)
Comment (chris-j-tang) [inline src/Commands/Parameters/StringLiteralParameter.ts]: After looking over the provided docs from the beginning of the semester, I don't know what you're referring to. Do I change the comment explaining what the `@param descriptor` refers to? Is there somewhere I reference to copy comments from?

Comment (JoshuaKGoldberg) [inline src/Commands/Parameters/StringLiteralParameter.ts]: By docs I meant the documentation comments you're writing here. You put `descriptor` and `parameters` as the arguments in your comments, but the actual argument names are `name` and `description`.

**Pull request merged**